### PR TITLE
applications: asset_tracker_v2: Cleanup P-GPS integration

### PR DESCRIPTION
**Pull request by tokangas**
Moved P-GPS request handling from data module to cloud module. Simplified P-GPS request handling and aligned P-GPS and A-GNSS implementations.

Added a missing value to cloud module backoff delay table.

Did a couple of minor documentation fixes.

NCSDK-18093

### CODE REVIEW

**Review by peknis — APPROVED** [review]

**Review by simensrostad — APPROVED** [review]
LGTM

File: applications/asset_tracker_v2/src/modules/cloud_module.c [hunk 1]
```diff
@@ -1048,53 +1252,17 @@ static void on_all_states(struct cloud_msg_data *msg)
 		}
 	}
 
+#if defined(CONFIG_NRF_CLOUD_AGNSS)
+	if (IS_EVENT(msg, location, LOCATION_MODULE_EVT_AGNSS_NEEDED)) {
```
Comment (trantanen) [inline applications/asset_tracker_v2/src/modules/cloud_module.c:1256]: There is some handling of `LOCATION_MODULE_EVT_AGNSS_NEEDED` event both in `on_all_states()` and `on_sub_state_cloud_disconnected()`. This is probably wrong and might be a bit more complicated to handle this perfectly as there is no obvious good place for either code piece. The code can obviously be copied into multiple states but hopefully we are able to find a proper place for it.

**Comment by tokangas** [comment]
Added the DNM label because I need to make changes to `LOCATION_MODULE_EVT_AGNSS_NEEDED` event handling as pointed out by @trantanen.

**Comment by tokangas** [comment]
> Added the DNM label because I need to make changes to `LOCATION_MODULE_EVT_AGNSS_NEEDED` event handling as pointed out by @trantanen.

Done, removing DNM.

File: applications/asset_tracker_v2/src/modules/cloud_module.c (hunk 1 above)
Comment (tokangas) [reply applications/asset_tracker_v2/src/modules/cloud_module.c]: Moved both `LOCATION_MODULE_EVT_AGNSS_NEEDED` and `LOCATION_MODULE_EVT_PGPS_NEEDED` handling from `on_all_states()` to `on_sub_state_cloud_connected()`.

When not connected to the cloud, A-GNSS request is buffered in `on_sub_state_cloud_disconnected()` and `on_state_lte_disconnected()`. As before, the buffered request is handled when moving from cloud disconnected to cloud connected state.

Comment (trantanen) [reply applications/asset_tracker_v2/src/modules/cloud_module.c]: Yep, this should be fine.

**Review by trantanen — APPROVED** [review]

**Pull request merged**